src/Controls/src/Core/HandlerImpl/Window/Window.Impl.cs

```diff
@@ -534,11 +534,15 @@ FlowDirection IWindow.FlowDirection
 				// Then we want the window to also reflect RTL
 				// We don't want to force the user to reach into the window
 				// in order to enable RTL Window features on WinUI
-				if (FlowDirection == FlowDirection.MatchParent &&
-					Page is IFlowDirectionController controller &&
-					controller.EffectiveFlowDirection.IsExplicit())
+				IFlowDirectionController? pageController = Page;
+				IFlowDirectionController currentPageController = Page.GetCurrentPage();
```

Comment:
One of them has a funny nullability annotation, but has nothing to do with you, has to do with the fact that the extension method is not doing the correct thing. If we look at:
```csharp
                internal static Page GetCurrentPage(this Page currentPage)
                {
                        if (currentPage is FlyoutPage fp)
                                return GetCurrentPage(fp.Detail);
                        else if (currentPage is IPageContainer<Page> pc)
                                return GetCurrentPage(pc.CurrentPage);
                        else
                                return currentPage;
                }
```
The file does not have nullability enabled AND has an issue. The implementation above shows that we can have a currentPageController that is nullable, because Page is Nullable. Evil extension methods. That extension should be fixed.